### ISSUE TITLE
Selection is not properly set after removal of last character in a text node

### ISSUE THREAD

**Issue by sashless** [facebook/draft-js#1257]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Imagine: A|Entity|B|Entity|C

When removing character 'A' or 'B' the cursor jumps to the end of the line. In this case character 'C'

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. You can use this jsfiddle to get started: https://jsfiddle.net/stopachka/m6z0xn4r/.**

Demo: https://jsfiddle.net/less/m6z0xn4r/224/

**What is the expected behavior?**

The cursor should remain at the current location after removing character 'A' or 'B'. 

**Which versions of Draft.js, and which browser / OS are affected by this issue? Did this work in previous versions of Draft.js?**

My demo is using 0.10.1 but its also occuring in 0.10.0. 
The problem seems to be that after removal of the character the selection is no longer inside a text node.
Basically it lands here because anchor and focus is no longer a text node
https://github.com/facebook/draft-js/blob/213cd764f61cc64552bddd672ae1748529d55333/src/component/selection/getDraftEditorSelectionWithNodes.js#L61

Then some internal draftjs code is looking for the nearest children to place it properly.

**Comment by mzedeler** [comment]
Similar issue: #1749. I think it is necessary to write a modifier that works on the `editorState` object, not just `contentState`.
